curvefs/src/client/extent_manager.h

```diff
@@ -33,74 +33,113 @@
- * @param[in,out] extents  extents in the file
- *
- * @return  errcode
- */
-CURVEFS_ERROR MergeAllocedExtents(
-    const std::list<ExtentAllocInfo> &toAllocExtents,
-    const std::list<Extent> allocatedExtents,
-    VolumeExtentList *extents);
-
-/**
- * @brief mark unwritten extents to written in the file extents
- *
- * @param[in] offset  offset to be written
- * @param[in] len  len to be written
- * @param[in,out] extents  extents in the file
- *
- * @return  errcode
- */
-CURVEFS_ERROR MarkExtentsWritten(uint64_t offset, uint64_t len,
-    VolumeExtentList *extents);
-
-/**
- * @brief According to the extents in the file,
- * divide the [lOffset, len] to be read/write,
- * and get the physical extents
- *
- * @param extents  extents in the file
- * @param offset  logical offset to be read/write
- * @param len  len to be read/write
- * @param PExtent  physical extents
- *
- * @return errcode
- */
-CURVEFS_ERROR DivideExtents(const VolumeExtentList &extents,
-    uint64_t lOffset,
-    uint64_t len,
-    std::list<PExtent> *pExtents);
-
-
+class ExtentManager {
+ public:
+    ExtentManager() {}
+    virtual ~ExtentManager() {}
+
+    /**
+     * @brief  According to the extents in the file,
+     * divide the [offset, len] to be written,
+     * and get the extents that need to be allocated
+     *
+     * @param[in] extents  extents in the file
+     * @param[in] offset  offset to be written
+     * @param[in] len  len to be written
+     * @param[out] toAllocExtents  extents that need to be allocated
+     *
+     * @return errcode
+     */
+    virtual CURVEFS_ERROR GetToAllocExtents(const VolumeExtentList &extents,
+        uint64_t offset,
+        uint64_t len,
+        std::list<ExtentAllocInfo> *toAllocExtents) = 0;
+
+    /**
+     * @brief  Merge extents allocated to extents in the file
+     *
+     * @param[in] allocedExtents  extents allocated
+     * @param[in] toAllocExtents  extents that need to be allocated
+     * @param[in,out] extents  extents in the file
+     *
+     * @return  errcode
+     */
+    virtual CURVEFS_ERROR MergeAllocedExtents(
+        const std::list<ExtentAllocInfo> &toAllocExtents,
+        const std::list<Extent> allocatedExtents,
+        VolumeExtentList *extents) = 0;
+
+    /**
+     * @brief mark unwritten extents to written in the file extents
+     *
+     * @param[in] offset  offset to be written
+     * @param[in] len  len to be written
+     * @param[in,out] extents  extents in the file
+     *
+     * @return  errcode
+     */
+    virtual CURVEFS_ERROR MarkExtentsWritten(uint64_t offset, uint64_t len,
+        VolumeExtentList *extents) = 0;
+
+    /**
+     * @brief According to the extents in the file,
+     * divide the [lOffset, len] to be read/write,
+     * and get the physical extents
+     *
+     * @param extents  extents in the file
+     * @param offset  logical offset to be read/write
+     * @param len  len to be read/write
+     * @param PExtent  physical extents
```

Comment:
we have logical extents?

Reply:
no

Reply:
logical extent  composed of physical extent list?

Reply:
they are different things，logical extent mean extent in the file, physical extent means extent in the block device.